### PR TITLE
Book: First draft of Integration testing

### PR DESCRIPTION
**Pull request by killercup**
cf #72

### CODE REVIEW

File: src/tutorial/testing.md [hunk 1]
```diff
@@ -271,15 +284,114 @@ can help you structure and organize your code.
...
+(and that have great error messages).
+
+[`assert_cmd`]: https://docs.rs/assert_cmd
+[`predicates`]: https://docs.rs/predicates
+
+This sounds like a lot of setup.
+Nevertheless --
+let's dive right in!
+
+```rust
```
Comment (Dylan-DPC) [inline src/tutorial/testing.md]: perhaps a good idea to mention where the user should write this code

File: src/tutorial/testing.md [hunk 2]
```diff
@@ -220,8 +233,8 @@ by adding [`&mut std::io::stdout()`][stdout] as the third parameter.
 
 </aside>
 
-We've just seen how to be make it easily testable,
-we have
+We've just seen how to be make this piece of code easily testable.
```
Comment (waywardmonkeys) [inline src/tutorial/testing.md]:
```suggestion
We've just seen how to make this piece of code easily testable.
```

File: src/tutorial/testing.md [hunk 3]
```diff
@@ -271,15 +284,114 @@ can help you structure and organize your code.
...
+`cargo` will look for integration tests in the `tests/` directory.
+Similarly,
+it will look for benchmarks in `benches/`,
+and examples in `examples`/.
+These conventions also extend to your main source code:
+libraries have a `src/lib.rs` file,
+the main binary is `src/main.rs`,
+or, if there are multiple binaries,
+cargo expects them to be in `src/bin/<name>.rs`.
+Following these conventions will make your code base easily discoverable
```
Comment (waywardmonkeys) [inline src/tutorial/testing.md]:
```suggestion
Following these conventions will make your code base more discoverable
```

File: src/tutorial/testing.md (hunk 1 above)
Comment (killercup) [reply src/tutorial/testing.md]: Hm, I wrote `tests/cli.rs` a few paragraphs above. Let me repeat that here.

**Comment by killercup** [comment]
Updated, and merging. Futher comments welcome in #58 :)

**Comment by killercup** [comment]
Oh -- @epage I meant to ping you on this one, too! (But I forgot to ping _anyone_…)

You can see the current version at the bottom of <https://rust-lang-nursery.github.io/cli-wg/tutorial/testing.html>. Would be interested in hearing if you have suggestions/ideas what to talk about here, too :)

**Comment by epage** [comment]
I saw the PR come through and looked back over it.  It seems reasonable.

**Pull request merged**